src/Microsoft.Azure.WebJobs.Host/InvocationFilterAttribute.cs

```diff
@@ -0,0 +1,39 @@
+﻿// InvocationFilterAttribute.cs
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// This is the first iteration for the Invocation Filter Attribute
+    /// </summary>
+    [CLSCompliant(false)]
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, Inherited = true, AllowMultiple = false)]
+    public abstract class InvocationFilterAttribute : Attribute
```

Comment:
>InvocationFilterAttribute [](start = 26, length = 25)

What's the invariant around state here? Does each method instance get its own copy of this attribute? And such, can the OnExecuting set instance methods that are accessed in OnExecuted?